network/wsNetwork_test.go

```diff
@@ -656,6 +656,29 @@ func TestWebsocketNetworkNoAddress(t *testing.T) {
 	}
 }
 
+func TestWebsocketNetworkNoGossipService(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	config := defaultConfig
+	config.EnableGossipService = false
+	netA := makeTestWebsocketNodeWithConfig(t, config)
```

Comment:
maybe check NetAddress = "" + EnableGossipService = true also do not accept incoming (i.e. EnableGossipService does not override NetAddress)